### ISSUE TITLE
Connecting device removes custom device configs from user device config

### ISSUE THREAD

**Issue by Renpona** [intiface/intiface-central#121]
If any custom device configurations (i.e., ones containing custom messages) have been added by the user to Intiface Central's buttplug-user-device-config.json file, then they will be removed by Intiface Central whenever a device is connected.

Specifically, the "configurations" section of the "specifiers" section gets wiped, and the "messages" section in the device entry gets completely removed.

### Steps to Reproduce:

1. Add a custom device configuration including a new messages config to Intiface Central's buttplug-user-device-config.json file.
2. Run Intiface Central.
3. Connect any device to Intiface Central.
4. Check buttplug-user-device-config.json again - it will have been overwritten by Intiface Central.

Here is an example file before connecting a device:
```{
  "version": {
    "major": 2,
    "minor": 0
  },
  "user-configs": {
    "specifiers": {
      "tcode-v03": {
        "websocket": {
          "names": [
            "TestDevice"
          ]
        },
        "configurations": [
          {
            "identifier": [
              "T1234"
            ],
            "name": "TestDevice",
            "messages": {
              "ScalarCmd": [
                {
                  "StepRange": [
                    0,
                    100
                  ],
                  "ActuatorType": "Vibrate"
                }
              ]
            }
          }
        ]
      }
    },
    "devices": [
      {
        "identifier": {
          "address": "T1234",
          "protocol": "tcode-v03",
          "identifier": "TestDevice"
        },
        "config": {
          "index": 0,
          "messages": {
            "ScalarCmd": [
              {
                "StepRange": [
                  0,
                  100
                ],
                "ActuatorType": "Vibrate"
              }
            ]
          }
        }
      }
    ]
  }
}
```

Here is what that same example file looks like after carrying out the steps to reproduce:
```{
  "version": {
      "major": 2,
      "minor": 0
  },
  "user-configs": {
      "specifiers": {
          "tcode-v03": {
              "websocket": {
                  "names": [
                      "TestDevice"
                  ]
              },
              "configurations": []
          }
      },
      "devices": [
          {
              "identifier": {
                  "address": "T1234",
                  "protocol": "tcode-v03",
                  "identifier": "TestDevice"
              },
              "config": {
                  "index": 0
              }
          }
      ]
  }
}
```

I was told in Discord that this will be fixed with the configuration system rewrite, but I'm opening the issue anyway just so it's tracked.

**Comment by Renpona** [comment]
Will likely be fixed by https://github.com/buttplugio/buttplug/issues/616

**Comment by qdot** [comment]
Yup. Not gonna make it into 2.5.6 as I need to get this release out ASAP, but hopefully 2.5.7

**Comment by Renpona** [comment]
Fixed in Intiface Central 2.6!
